City-of-Helsinki/drupal-helfi-rekry
# UHF-7410: Language related accessibility fixes for job listings and job search

# [UHF-7410](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7410)

This PR adds correct wrapping elements to lang parameters on job listing pages (spans for inline elements) and separates the language wrapper from the the translated "x jobs" text in job search results.

## How to install
* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-7410-accessibility-language-fixes`
  * `make fresh`
  * If you get an ERROR 1273 (HY000) at line 14784: Unknown collation: 'utf8mb4_0900_ai_ci' error with the database import, run `make shell` and then:
    * `sed -i 's/utf8mb4_0900_ai_ci/utf8mb4_swedish_ci/g' dump.sql; drush sql-query --file=/app/dump.sql && drush helfi:oc:sanitize-database && drush cim -y;drush updb -y;drush cr;drush uli`
* Run `make ps` to get the correct proxy port for `drupal-helfi-rekry-elastic`. Should be 9200 but might change if you have multiple ES proxies running.
  * Edit the `.env` file and change the port in `ELASTIC_PROXY_URL`
  * Run `make up` again for the change to take effect
* Run `make drush-cr`
* Run `make shell` and then `drush sapi-rt;drush sapi-i` to make sure content is indexed properly 

## How to test
* [ ] Job search result changes work as intended
  * Go to https://helfi-rekry.docker.so/en/open-jobs/find-open-jobs and find a listing that has multiple jobs
  * Inspect the heading element. The untranslated part should be wrapped with `<span lang>` element while the "x job" text should be wrapped with just a blank `<span>` element.
  * Find a listing that doesn't have multiple jobs. Inspect the heading element to make sure it doesn't have an empty `<span>` after the heading.
* [ ] Job listing page changes work as intended
  *  On the english search page, find a finnish language job listing and open it
  * Inspect the heading and the salary field. It should be wrapped in a span element with a language parameter instead of a div
  * Inspect the body text. The p-elements should be contained in a div that has the language parameter

## Designers review


[UHF-7410]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-7410?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
